### PR TITLE
Modernize the maven build

### PR DESCRIPTION
**Pull request by nedtwigg**
Adopt the excellent [`maven-plugin-development`](https://github.com/britter/maven-plugin-development) plugin by @britter. Closes

- #554

### CODE REVIEW

File: plugin-maven/build.gradle [hunk 1]
```diff
@@ -1,42 +1,15 @@
...
 version = spotlessChangelog.versionNext
-apply from: rootProject.file("gradle/java-setup.gradle")
-apply from: rootProject.file("gradle/java-publish.gradle")
 
-final MAVEN_PROJECT_DIR = project.layout.buildDirectory.dir("mavenProject").get()
-final LOCAL_MAVEN_REPO_DIR = project.layout.buildDirectory.dir("localMavenRepository").get()
+apply from: rootProject.file("gradle/java-setup.gradle")
+apply from: rootProject.file('gradle/spotless-freshmark.gradle')
 
 def mvnw(String args) {
```
Comment (lutovich) [inline plugin-maven/build.gradle:14]: Is it now possible to remove this method?

File: plugin-maven/build.gradle [hunk 2]
```diff
@@ -55,17 +28,20 @@ def mvnw(String args) {
...
+apply plugin: 'de.benediktritter.maven-plugin-development'
+mavenPlugin {
+	name = 'Spotless Maven Plugin'
+	artifactId = project.artifactIdMaven
+	description = project.description
+}
+
+String VER_MAVEN_API = '3.0'
+String VER_ECLIPSE_AETHER = '1.1.0'
+String VER_MUSTACHE = '0.9.10'
```
Comment (lutovich) [inline plugin-maven/build.gradle:40]: Perhaps inline this constant? It is only used for a single `testImplementation` dependency

**Review by lutovich — APPROVED** [review]
LGTM! Added two very minor comments/questions

File: plugin-maven/build.gradle (hunk 1 above)
Comment (nedtwigg) [reply plugin-maven/build.gradle]: Good call! Done.

File: plugin-maven/build.gradle (hunk 2 above)
Comment (nedtwigg) [reply plugin-maven/build.gradle]: Good call! Done.

**Comment by nedtwigg** [comment]
Released in `plugin-maven 2.31.0`.

**Pull request merged**